app/src/main/java/com/marvelapp/presentation/activity/MainMenuActivity.kt

```diff
@@ -31,16 +32,23 @@ class MainMenuActivity : AppCompatActivity() {
         lifecycleScope.launchWhenStarted {
             viewModel.state.collectLatest { this@MainMenuActivity.updateUI(it) }
         }
+        binding.mainMenuButton.setOnClickListener { viewModel.buttonPressed() }
     }
 
     private fun updateUI(state: MainMenuViewModel.MainMenuStates) {
         when (state) {
             INIT -> {
                 // DO NOTHING
             }
+            GO_TO_CHARACTER_SCREEN -> {
+                goToCharacterScreen()
+            }
         }
     }
 
+    private fun goToCharacterScreen() {
+        startActivity(CharacterActivity.getIntent(this))
```

Comment:
you can avoid creating this method and put this line in the correspondent option of the when statement

Reply:
Solved